cli/templateedit.go

```diff
@@ -297,3 +235,170 @@ func (r *RootCmd) templateEdit() *clibase.Cmd {
 
 	return cmd
 }
+
+type checkEditTemplateEntitlementsArgs struct {
+	client *codersdk.Client
+	inv    *clibase.Invocation
+
+	defaultTTL                     time.Duration
+	maxTTL                         time.Duration
+	autostopRequirementDaysOfWeek  []string
+	autostopRequirementWeeks       int64
+	autostartRequirementDaysOfWeek []string
+	failureTTL                     time.Duration
+	dormancyThreshold              time.Duration
+	dormancyAutoDeletion           time.Duration
+	allowUserCancelWorkspaceJobs   bool
+	allowUserAutostart             bool
+	allowUserAutostop              bool
+	requireActiveVersion           bool
+}
+
+func checkEditTemplateEntitlements(ctx context.Context, args checkEditTemplateEntitlementsArgs) (bool, error) {
```

Comment:
A bit surprised that we can't reuse much of the code between checking edit and checking create entitlements?